### PR TITLE
SUS-358: Send token and forward client IP to MW

### PR DESCRIPTION
**Pull request by wladekb**
This change will ensure we send token when we should and will also forward the client IP so MW can properly do checks.

https://wikia-inc.atlassian.net/browse/SUS-358

/cc @pawelch @dianafa @owend

### CODE REVIEW

File: server_config.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ default_config.MAX_MESSAGES_IN_BACKLOG = 50
 // how many messages each room will store for now. only longer than NUM_MESSAGES_TO_SHOW_ON_CONNECT for potential debugging.
 default_config.NUM_MESSAGES_TO_SHOW_ON_CONNECT = 10
 default_config.TOKEN = "Token"
-default_config.WIKIA_CLIENT_IP_HEADER = "X-Forwarded-For"
+default_config.TRUSTED_CLIENT_IP_HEADER = "" // lowercase
```
Comment (sactage) [inline server_config.js]: what does the `// lowercase` comment mean here?

Comment (wladekb) [inline server_config.js]: Fixed.

File: WMBridge.js [hunk 2]
```diff
@@ -73,9 +74,17 @@ var requestMW = function(method, roomId, postdata, query, handshake, callback, e
...
+
+				if (config.TRUSTED_CLIENT_IP_HEADER) {
+					if (typeof handshake.headers[config.TRUSTED_CLIENT_IP_HEADER] !== 'undefined') {
+						// forward header if we got one
+						headers[config.TRUSTED_CLIENT_IP_HEADER] = handshake.headers[config.TRUSTED_CLIENT_IP_HEADER];
+					} else if (headers['x-forwarded-for']) {
+						// make sure MW uses this client IP - even if chat server isn't on whitelisted IP
+						// see $wgSquidServersNoPurge in MW config
+						headers[config.TRUSTED_CLIENT_IP_HEADER] = headers['x-forwarded-for'].split(',').shift();
+					}
```
Comment (dianafa) [inline WMBridge.js]: we are using here `config.TRUSTED_CLIENT_IP_HEADER` 5 times, and `handshake.headers[config.TRUSTED_CLIENT_IP_HEADER]` twice, can we move it to a separate vars?

File: server_config.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ default_config.MAX_MESSAGES_IN_BACKLOG = 50
 // how many messages each room will store for now. only longer than NUM_MESSAGES_TO_SHOW_ON_CONNECT for potential debugging.
 default_config.NUM_MESSAGES_TO_SHOW_ON_CONNECT = 10
 default_config.TOKEN = "Token"
-default_config.WIKIA_CLIENT_IP_HEADER = "X-Forwarded-For"
```
Comment (dianafa) [inline server_config.js]: wasn't this var used anywhere?

Comment (sactage) [inline server_config.js]: It doesn't look like it, no. I surmise it was meant to be used around https://github.com/Wikia/chat/pull/25/files#diff-ca1bdb770ba8bf8292cccfa249a49638L78 - but that's only a guess.

**Comment by wladekb** [comment]
Now part of #26

**Pull request closed without merging**